OpenDreamRuntime/Objects/Types/DreamObjectSavefile.cs

```
@@ -150,31 +164,372 @@ public sealed class SavefileDirectory : Dictionary<string, DreamValue> { }
+                        if(SFDreamListValue.AssocData != null && SFDreamListValue.AssocData[i] != null) //note that null != DreamValue.Null
+                            l.SetValue(DeserializeJsonValue(SFDreamListValue.AssocKeys[i]), DeserializeJsonValue(SFDreamListValue.AssocData[i]!));
+                        else
+                            l.AddValue(DeserializeJsonValue(SFDreamListValue.AssocKeys[i]));
+                    }
+                }
+                return new DreamValue(l);
+            case SFDreamObjectPathValue SFDreamObjectPath:
+                SFDreamJsonValue storedObjectVars = SFDreamObjectPath;
+                SFDreamJsonValue searchDir = SFDreamObjectPath;
+                while(searchDir != _rootNode)
+                    if(!searchDir.TryGetValue(SFDreamObjectPath.Path, out storedObjectVars!))
+                        searchDir = SeekTo("..");
+                    else
+                        break;
+
+                if(storedObjectVars!.TryGetValue("type", out SFDreamJsonValue? storedObjectTypeJSON) && DeserializeJsonValue(storedObjectTypeJSON).TryGetValueAsType(out TreeEntry? objecTypeActual)) {
+                    DreamObject resultObj = _objectTree.CreateObject(objecTypeActual);
+                    foreach(string key in storedObjectVars.Keys){
+                        if(key == "type")
+                            continue;
+                        resultObj.SetVariable(key, DeserializeJsonValue(storedObjectVars[key]));
+                    }
+                    resultObj.InitSpawn(new DreamProcArguments());
+                    resultObj.SpawnProc("Read", null, [new DreamValue(this)]);
+                    return new DreamValue(resultObj);
+                } else
+                    throw new InvalidDataException("Unable to deserialize object in savefile: " + ((storedObjectTypeJSON as SFDreamType) is null ? "no type specified (corrupted savefile?)" : "invalid type "+((SFDreamType)storedObjectTypeJSON!).TypePath));
```

silently errors apparently
https://discord.com/channels/847318834952405042/887074737879138315/1200283136026755152

---

well that's insane. How the hell is it inferring the type if type is overridden?
Honestly think this one should be an error.

---

Turns out the test case was bad, `F << S["foo"]` is a no-op. BYOND just silently fails and outputs a null. We choose to fail loudly instead.